### DISCUSSION TITLE
Use SSH keys for private github repository

### DISCUSSION THREAD

**Discussion by bastien70** [deployphp/deployer#2308] [Help needed]
**Deployer version : v6.8.0**

Hello, I am trying to use deployer to deploy a private github repository.

After various research, this is what I did:

I added a deployment key in my github repository (the same one I use to connect in SSH to my o2Switch server with Deployer)

In my "deploy.php" file, I changed the way to declare my repository, to use SSH:

`git@github.com:bastien70/tuto-deployer.git`

But when I launch the **dep deploy -vvv** command, when it tries to clone the repository, this is what I have:

![git1](https://user-images.githubusercontent.com/53140475/100519255-15c53300-3197-11eb-8d21-ea23af3f414f.PNG)
![git2](https://user-images.githubusercontent.com/53140475/100519254-152c9c80-3197-11eb-8ada-d65fb3cabaf0.PNG)

**Comment by antonmedv** [comment]
Use deploy keys for this.

**Reply by bastien70** [reply]
This is what I did, I put my public key as "deploy key", but I don't understand what to do next

EDIT : Ok I resolved the problem. The `ssh -T git@github.com` was giving me an error. I had to create a config file, like this : 

```
Host github.com
 Hostname ssh.github.com
 Port 443
 PreferredAuthentications publickey
 IdentityFile ~/.ssh/github/id_rsa
```

**Comment by getafixx** [comment]
but this doesn't work if you have two reports that you want to pull from. 

I have tried to get two deploys from two different repos (same owner) to use two different public keys and it just will not work.

Has anyone else had this issue?

**Reply by antonmedv** [reply]
Why not? Add to different deploy keys.
